File: app/src/main/java/com/bracketcove/postrainer/alarmdetail/AlarmDetailContract.java
package com.bracketcove.postrainer.alarmdetail;

import com.bracketcove.postrainer.base.BasePresenter;
import com.bracketcove.postrainer.base.BaseView;
import com.bracketcove.postrainer.data.viewmodel.Alarm;

/**
 * Created by Ryan on 06/03/2017.
 */

public interface AlarmDetailContract {
    interface View extends BaseView {
        Alarm getViewModel();

        void setAlarmTitle(String title);

        void setVibrateOnly(boolean active);

        void setRenewAutomatically(boolean active);

        void setPickerTime(int hour, int minute);

        void setCurrentAlarmState(boolean active);

        String getAlarmId();

        void startAlarmListActivity();

    }

    interface Presenter extends BasePresenter {
        void onBackIconPress();

        void onDoneIconPress();
    }
}

File: app/src/main/java/com/bracketcove/postrainer/alarm/AlarmDetailContract.kt
package com.bracketcove.postrainer.alarm

import com.wiseassblog.domain.domainmodel.Alarm

/**
 * Created by Ryan on 06/03/2017.
 */

interface AlarmDetailContract {
    interface View  {
        fun setAlarmTitle(title:String)

        fun setVibrateOnly(vibrateOnly: Boolean)

        fun setRenewAutomatically(renewAutomatically: Boolean)

        fun setPickerTime(hour: Int, minute: Int)

        fun startAlarmListActivity()

        fun showMessage(msg: String)

        fun showDeleteConfirm()

        fun getState(): Alarm
    }

    interface ViewModel {
        fun getAlarm(): Alarm?
        fun setAlarm(alarm: Alarm)
    }
}

sealed class AlarmDetailEvent {
    object OnBackIconPress : AlarmDetailEvent()
    object OnDoneIconPress : AlarmDetailEvent()
    object OnDeleteIconPress : AlarmDetailEvent()
    object OnDeleteConfirmed : AlarmDetailEvent()
    object OnStart : AlarmDetailEvent()
}
